### PR TITLE
fix(gatsby-plugin-image): Conditionally require dependencies and give better warnings

### PR DESCRIPTION
**Pull request by laurieontech**
Right now gatsby-plugin-image will break builds if gatsby-source-filesystem and gatsby-plugin-sharp are not included as dependencies. 

This should not happen if filesystem isn't there as it's only needed for StaticImage, not GatsbyImage.

plugin-sharp should error, but with a more descriptive error than the generic webpack one. [ch22339]

### CODE REVIEW

File: packages/gatsby-plugin-image/src/node-apis/image-processing.ts [hunk 1]
```diff
@@ -75,6 +85,14 @@ export async function writeImages({
       let file: FileSystemNode | undefined
       let fullPath
       if (process.env.GATSBY_EXPERIMENTAL_REMOTE_IMAGES && isRemoteURL(src)) {
+        let createRemoteFileNode
+        try {
+          createRemoteFileNode = require(`gatsby-source-filesystem`)
```
Comment (ascorbic) [inline packages/gatsby-plugin-image/src/node-apis/image-processing.ts:90]: This needs to be destructured

**Comment by ascorbic** [comment]
Shouldn't these all be `reporter.panic`, as the missing dependencies are fatal errors?

File: packages/gatsby-plugin-image/src/node-apis/image-processing.ts (hunk 1 above)
Comment (laurieontech) [reply packages/gatsby-plugin-image/src/node-apis/image-processing.ts:90]: Ooh good catch, missed this one.

**Comment by laurieontech** [comment]
@ascorbic error seemed to work in my testing but if panic is the appropriate function I'll certainly use that instead.

**Review by ascorbic — APPROVED** [review]
Looks great!

**Pull request merged**